File: frontend/src/features/analytics/services/analytics.ts
import api from "@/services/api";

import type {
  AnalyticsSummary,
  SeverityDistribution,
  ThreatTrend,
  OSDistribution,
  EnvironmentDistribution,
  AssetTypeDistribution,
  SecurityRecommendation,
  DiscoverySummary,
} from "../types/analytics";

export async function getAnalytics(): Promise<AnalyticsSummary> {
  const response = await api.get("/analytics/overview");

  return {
    totalAlerts: response.data.total_alerts,
    criticalAlerts: response.data.critical_alerts,
    totalAssets: response.data.total_assets,
    totalScans: response.data.total_scans,
    averageRiskScore: response.data.average_risk_score,
  };
}

export async function getSeverityDistribution(): Promise<
  SeverityDistribution[]
> {
  const response = await api.get(
    "/dashboard/severity-distribution",
  );

  return response.data;
}

export async function getThreatTrend(): Promise<
  ThreatTrend[]
> {
  const response = await api.get(
    "/dashboard/threat-trend"
  );

  return response.data;
}

import type { TopRiskAsset } from "../types/analytics";

export async function getTopRiskAssets(): Promise<
  TopRiskAsset[]
> {
  const response = await api.get(
    "/analytics/top-risk-assets"
  );

  return response.data;
}

export async function getOSDistribution(): Promise<
  OSDistribution[]
> {
  const response = await api.get(
    "/analytics/os-distribution"
  );

  return response.data;
}

export async function getEnvironmentDistribution(): Promise<
  EnvironmentDistribution[]
> {
  const response = await api.get(
    "/analytics/environment-distribution"
  );

  return response.data;
}

export async function getAssetTypeDistribution(): Promise<
  AssetTypeDistribution[]
> {
  const response = await api.get(
    "/analytics/asset-type-distribution"
  );

  return response.data;
}

export async function getSecurityRecommendations(): Promise<
  SecurityRecommendation[]
> {
  const response = await api.get(
    "/analytics/security-recommendations"
  );

  return response.data;
}

export async function getDiscoverySummary(): Promise<
  DiscoverySummary
> {
  const response = await api.get(
    "/analytics/discovery-summary"
  );

  return response.data;
}

File: backend/app/api/v1/endpoints/analytics.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.database.session import get_db
from app.services.analytics_service import AnalyticsService
from app.schemas.analytics import AnalyticsResponse

from app.schemas.asset import AssetResponse

from app.schemas.recommendation import (
    SecurityRecommendation,
)

from app.schemas.discovery_summary import (
    DiscoverySummaryResponse,
)

router = APIRouter(
    prefix="/analytics",
    tags=["Analytics"],
)


@router.get(
    "/overview",
    response_model=AnalyticsResponse,
)
def get_analytics(
    db: Session = Depends(get_db),
):
    service = AnalyticsService(db)
    return service.get_overview()


@router.get(
    "/top-risk-assets",
    response_model=list[AssetResponse],
)
def get_top_risk_assets(
    db: Session = Depends(get_db),
):
    service = AnalyticsService(db)
    return service.get_top_risk_assets()


@router.get("/os-distribution")
def get_os_distribution(
    db: Session = Depends(get_db),
):
    service = AnalyticsService(db)
    return service.get_os_distribution()


@router.get("/environment-distribution")
def get_environment_distribution(
    db: Session = Depends(get_db),
):
    service = AnalyticsService(db)
    return service.get_environment_distribution()


@router.get("/asset-type-distribution")
def get_asset_type_distribution(
    db: Session = Depends(get_db),
):
    service = AnalyticsService(db)
    return service.get_asset_type_distribution()


@router.get(
    "/security-recommendations",
    response_model=list[SecurityRecommendation],
)
def get_security_recommendations(
    db: Session = Depends(get_db),
):
    service = AnalyticsService(db)
    return service.get_security_recommendations()


@router.get(
    "/discovery-summary",
    response_model=DiscoverySummaryResponse,
)
def get_discovery_summary(
    db: Session = Depends(get_db),
):
    service = AnalyticsService(db)
    return service.get_discovery_summary()